lib/active_merchant/billing/gateways/pro_pay.rb

```diff
@@ -0,0 +1,327 @@
+        "61" => "Exceeds issuer withdrawal limit",
+        "62" => "Issuer does not allow this card to be charged for your business.",
+        "63" => "Security Violation",
+        "65" => "Activity limit exceeded",
+        "75" => "PIN tries exceeded",
+        "76" => "Unable to locate account",
+        "78" => "Account not recognized",
+        "80" => "Invalid Date",
+        "82" => "Invalid CVV2",
+        "83" => "Cannot verify the PIN",
+        "85" => "Service not supported for this card",
+        "93" => "Cannot complete transaction. Customer should call 800 number.",
+        "95" => "Misc Error Transaction failure",
+        "96" => "Issuer system malfunction or timeout.",
+        "97" => "Approved for a lesser amount. ProPay will not settle and consider this a decline.",
+        "98" => "Failure HV",
+        "99" => "Generic decline or unable to parse issuer response code"
+      }
+
+      def initialize(options={})
+        requires!(options, :cert_str)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        request = build_xml_request do |xml|
+          add_invoice(xml, money, options)
+          add_payment(xml, payment, options)
+          add_address(xml, options)
+          add_account(xml, options)
+          add_recurring(xml, options)
+          xml.transType "04"
+        end
+
+        commit(request)
+      end
+
+      def authorize(money, payment, options={})
+        request = build_xml_request do |xml|
+          add_invoice(xml, money, options)
+          add_payment(xml, payment, options)
+          add_address(xml, options)
+          add_account(xml, options)
+          add_recurring(xml, options)
+          xml.transType "05"
+        end
+
+        commit(request)
+      end
+
+      def capture(money, authorization, options={})
+        request = build_xml_request do |xml|
+          xml.amount money if money
+          xml.transNum authorization
+          add_account(xml, options)
+          xml.transType "06"
+        end
+
+        commit(request)
+      end
+
+      def refund(money, authorization, options={})
+        request = build_xml_request do |xml|
+          xml.transNum authorization
+          xml.amount money if money
+          xml.currencyCode if options[:currency]
+          add_account(xml, options)
+          xml.transType "07"
+        end
+
+        commit(request)
+      end
+
+      def void(authorization, options={})
+        refund(nil, authorization, options)
+      end
+
+      def credit(money, payment, options={})
+        request = build_xml_request do |xml|
+          add_invoice(xml, money, options)
+          add_payment(xml, payment, options)
+          add_account(xml, options)
+          xml.transType "35"
+        end
+
+        commit(request)
+      end
+
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
```

Comment:
Hmmm I didn't try it actually. I've actually always gone off of the `100` that comes with the generator. I'll give it a shot and see what happens.